cfme/tests/cloud/test_quota.py

```diff
@@ -298,3 +305,195 @@ def test_project_quota_enforce_via_lifecycle_cloud(
+                                "dynamic": True,
+                                'show_refresh_button': True,
+                                'load_values_on_init': True,
+                                'read_only': False,
+                                'auto_refresh': False,
+                                'visible': True,
+                                "type": "DialogFieldDropDownList",
+                                "resource_action": {
+                                    "resource_type": "DialogField",
+                                    "ae_namespace": automate_flavour_method.namespace.name,
+                                    "ae_class": automate_flavour_method.klass.name,
+                                    "ae_instance": automate_flavour_method.name,
+                                },
+                            },
+                        ],
+                    }
+                ],
+            }
+        ],
+    }
+    dialog_rest = appliance.rest_api.collections.service_dialogs.action.create(**data)[0]
+    yield appliance.collections.service_dialogs.instantiate(label=dialog_rest.label)
+    dialog_rest.action.delete()
+
+
+def get_quota_message(request, appliance, catalog, catalog_item_name, dialog_values=None):
+    """Returns the quota requested by particular type of flavor type"""
+    service_catalogs = ServiceCatalogs(appliance, catalog, catalog_item_name, dialog_values)
+    service_catalogs.order()
+    # nav to requests page to check quota validation
+    request_description = 'Provisioning Service [{0}] from [{0}]'.format(catalog_item_name)
+    provision_request = appliance.collections.requests.instantiate(request_description)
+    provision_request.wait_for_request(method='ui')
+    request.addfinalizer(provision_request.remove_request)
+    assert provision_request.row.reason.text == "Quota Exceeded"
+    last_message = provision_request.row.last_message.text
+    result = re.findall(r'requested.*\w', last_message)
+
+    # Service request needs to delete because we are not able to order same catalog item multiple
+    # times using automation.
+    delete_request = appliance.rest_api.collections.service_requests.get(
+        description=request_description)
+    delete_request.action.delete()
+
+    return result
+
+
+# first arg of parametrize is the list of fixture or parameter,
+# second arg is a list of lists, with a test is to be generated
+# indirect is the list where we define which fixture is to be passed values indirectly.
+@pytest.mark.meta(blockers=[BZ(1704439)])
+@pytest.mark.tier(1)
+@pytest.mark.parametrize(
+    ['set_roottenant_quota'],
+    [
+        [('storage', 0.001)]
+    ],
+    indirect=['set_roottenant_quota'],
+    ids=['max_storage']
+)
+def test_custom_service_dialog_quota_flavors(request, provider, provisioning, dialog, catalog,
+                                             appliance, automate_flavour_method, admin_email,
```

Comment:
Your fixture list is including some that don't need to be here, like automate_flavour_method. This test doesn't use that fixture directly, it will be called by `dialog` fixture.